tools/opt/opt.cpp

```diff
@@ -0,0 +1,160 @@
+// "Materials"), to deal in the Materials without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Materials, and to
+// permit persons to whom the Materials are furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Materials.
+//
+// MODIFICATIONS TO THIS FILE MAY MEAN IT NO LONGER ACCURATELY REFLECTS
+// KHRONOS STANDARDS. THE UNMODIFIED, NORMATIVE VERSIONS OF KHRONOS
+// SPECIFICATIONS AND HEADER INFORMATION ARE LOCATED AT
+//    https://www.khronos.org/registry/
+//
+// THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+#include "opt/ir_loader.h"
+#include "opt/libspirv.hpp"
+#include "opt/pass_manager.h"
+
+using namespace spvtools;
+
+void PrintUsage(const char* program) {
+  printf(
+      R"(%s - Optimize a SPIR-V binary file.
+
+USAGE: %s [options] [<input>] -o <output>
+
+The SPIR-V binary is read from <input>. If no file is specified,
+or if <input> is "-", then the binary is read from standard input.
+if <output> is "-", then the optimized output is written to
+standard output.
+
+NOTE: The optimizer is a work in progress.
+
+Options:
+  --strip-debug-info
+               Remove all debug instructions.
+  -h, --help   Print this help.
+  --version    Display optimizer version information.
+)",
+      program, program);
+}
+
+int main(int argc, char** argv) {
+  const char* in_file = nullptr;
+  const char* out_file = nullptr;
+
+  spv_target_env target_env = SPV_ENV_UNIVERSAL_1_1;
+
+  opt::PassManager pass_manager;
+
+  for (int argi = 1; argi < argc; ++argi) {
+    const char* cur_arg = argv[argi];
+    if ('-' == cur_arg[0]) {
+      if (0 == strcmp(cur_arg, "--version")) {
+        printf("%s\n", spvSoftwareVersionDetailsString());
+        return 0;
+      } else if (0 == strcmp(cur_arg, "--help") || 0 == strcmp(cur_arg, "-h")) {
+        PrintUsage(argv[0]);
+        return 0;
+      } else if (0 == strcmp(cur_arg, "-o")) {
+        if (!out_file && argi + 1 < argc) {
+          out_file = argv[++argi];
+        } else {
+          PrintUsage(argv[0]);
+          return 1;
+        }
+      } else if (0 == strcmp(cur_arg, "--strip-debug-info")) {
+        pass_manager.AddPass<opt::StripDebugInfoPass>();
+      } else if (0 == cur_arg[1]) {
```

Comment:
Nit: if you are comparing against a char you should probably use the '\0' instead of 0.